main/src/mill/main/MainModule.scala

```diff
@@ -173,4 +175,17 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets will clean everything.
+    */
+  def clean(targets: String*) = mill.T.command {
+    val result = targets match {
+      case Nil =>
+        List(ammonite.ops.%%('rm, "-rf", OutDir)(ammonite.ops.pwd).out.string.trim())
+      case _ =>
+        targets.map(cleanTarget => ammonite.ops.%%('rm, "-rf", s"$OutDir/$cleanTarget")(ammonite.ops.pwd).out.string.trim())
```

Comment:
This won't work for targets like `foo.bar.baz`

Reply:
Yeah I wanted to ask about this first to confirm that it is desirable. Will mark as WIP until I've added this.